clients/src/main/java/org/apache/kafka/common/network/SaslChannelBuilder.java

```diff
@@ -190,9 +191,13 @@ public Set<String> reconfigurableConfigs() {
     }
 
     @Override
-    public void validateReconfiguration(Map<String, ?> configs) {
+    public void validateReconfiguration(Map<String, ?> configs) throws ConfigException {
         if (this.securityProtocol == SecurityProtocol.SASL_SSL)
-            sslFactory.validateReconfiguration(configs);
+            try {
+                sslFactory.validateReconfiguration(configs);
+            } catch (IllegalStateException e) {
+                throw new ConfigException("SASL reconfiguration failed due to " + e);
```

Comment:
I'm concatenating the "SASL reconfiguration failed" message with the underlying exception string now since it looks like `ConfigException` will print errors in a very specific way